core/kubernetes/importer/core-openhim-config-importer-job.yaml

```diff
@@ -37,7 +37,7 @@ spec:
           volumeMounts:
             - mountPath: /importer
               name: core-openhim-config-importer-config
-      restartPolicy: Never
+      restartPolicy: OnFailure
```

Comment:
@rcrichton Why is this restart on OnFailure? Surely if it fails it means there's something wrong with the OpenHIM instance or we've supplied the wrong credentials. Either way, we would have triggered this manually and seen it fail. We would then need to redeploy the OpenHIM or the importer and in that time there would be no need to reattempt the importer job

Reply:
We want all the setup to be as automatic as possible. We don't want to someone to be watching the status of the job. I think it's better to retry on failure. I've had cases where this failed because something wasn't up yet and the retry helps in this case.

Reply:
Also, the ./instant.sh script executed both the normal and config script for you so the user might not be aware of needing to restart the job.

Reply:
Sorry, another point :) The end goal of this project is for a lay person to be able to spin up an HIE automatically by the click of a button. So, as much as we can automate and make failure resilient the better.

Reply:
Maybe just to jump in a bit. If the job fails for some reason it is most likely due to a valid reason in terms of not authenticating to the service its trying to import to.

The initContainer is used to verify that the service we are trying to reach is up and running and able to receive the request we are sending. This means the pod will not fail as it wont have started yet and will keep trying. 

If the pod fails at this point then its most likely the user credentials which is incorrect, or the import object isnt valid. Restarting the same job at this point will not have a different result as the creds or object will still be invalid

Reply:
What we maybe need is some kind of feedback to the user to inform them that the import failed.

Running the setup from scratch should not result in any errors as everything is setup vanilla. but if we trigger the entire setup script again it will fail on the importer due to the password having changed. If this job is set to restart it will never complete and therefor never complete the `instant.sh` startup process

Reply:
I ran into cases where restarting on failure was necessary, but I think this was before the init container got merged. I don't mind trying to change these to `restartPolicy: Never` if you guys think that is best and then we can see if we encounter scenarios where this is necessary again.

I'l make the change for the moment.

Reply:
I'm leaving the restart policy on the docker ones as we don't have a way to ensure those are up before these are run. Does that make sense to you?

Reply:
Ok, I tested this out and the problem that I run into is that the mapper-config job can sometimes try to run before the core config has been run. This causes it to fail because the root user password is not yet set. We don't have a good way to make sure that job has run yet. The `restartPolicy: OnFailure` was a good stop gap for this. Do you think it should stay for this reason or should we try find another approach?

Reply:
hmmm ok cool, i see how your scenario plays out. So the core API starts up which means `/heartbeat` is accessible but at this time the `root@openhim.org` hasnt been inserted into the DB yet, So the request to import fails as this user doesnt yet exist. 

In that case i think we need to include a backoff period, so that the pod is only restarted say maximum 3 times so that it cant get stuck in a failure state. 

We might also need to do a cleanup of the job once its completed to remove any pods it created that might be in an Error state.

https://kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/#pod-backoff-failure-policy

Reply:
Sorry realized that version of Kubernetes isnt yet supported on AWS.
https://v1-14.docs.kubernetes.io/docs/concepts/workloads/controllers/jobs-run-to-completion/

Reply:
Alright, I've set the backoffLimit to 3. I didn't want to run a clean up automatically as then we lose logs that will be useful for any debugging.

Reply:
Ok cool sounds good :+1: